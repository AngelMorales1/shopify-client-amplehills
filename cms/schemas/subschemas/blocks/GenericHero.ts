import image from './../image';
import color from './../color';
import portableText from './../portableText';

export default {
  type: 'object',
  name: 'genericHero',
  title: 'Generic Hero',
  fieldsets: [
    {
      name: 'options',
      title: 'Options',
      options: { collapsible: true, collapsed: true }
    },
    {
      name: 'buttons',
      title: 'Buttons',
      options: { collapsible: true, collapsed: true }
    }
  ],
  fields: [
    {
      name: 'title',
      title: 'Title',
      type: 'string'
    },
    portableText({
      name: 'text',
      title: 'Text'
    }),
    image({
      name: 'image',
      title: 'Image'
    }),
    {
      name: 'imagePosition',
      title: 'Image Position',
      type: 'string',
      options: {
        list: ['Stacked', 'Inline']
      }
    },
    color({
      name: 'backgroundColor',
      title: 'Background Color'
    }),
    {
      name: 'buttonLabel',
      title: 'Button Label',
      fieldset: 'buttons',
      type: 'string'
    },
    {
      name: 'buttonLink',
      title: 'Button Link',
      fieldset: 'buttons',
      type: 'url',
      validation: Rule => Rule.uri({ allowRelative: true })
    },
    {
      name: 'drip',
      title: 'Has Drip?',
      fieldset: 'options',
      type: 'boolean'
    },
    {
      name: 'reverse',
      title: 'Reverse Image/Text Arrangement?',
      fieldset: 'options',
      type: 'boolean'
    },
  ]
};
