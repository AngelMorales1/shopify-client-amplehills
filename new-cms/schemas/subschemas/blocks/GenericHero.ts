import image from './../image';
import color from './../color';
import portableText from './../portableText';

export default {
  type: 'object',
  name: 'genericHero',
  title: 'Generic Hero',
  fieldsets: [
    {
      name: 'options',
      title: 'Options',
      options: { collapsible: true, collapsed: true }
    },
    {
      name: 'buttons',
      title: 'Buttons',
      options: { collapsible: true, collapsed: true }
    },
    {
      name: 'image',
      title: 'Image',
      options: { collapsible: true, collapsed: true }
    }
  ],
  fields: [
    {
      name: 'title',
      title: 'Title',
      type: 'string'
    },
    portableText({
      name: 'text',
      title: 'Text'
    }),
    color({
      name: 'backgroundColor',
      title: 'Background Color'
    }),
    image({
      name: 'image',
      title: 'Image',
      fieldset: 'image'
    }),
    {
      name: 'imagePosition',
      title: 'Image Position',
      type: 'string',
      options: {
        list: ['Stacked', 'Inline']
      },
      fieldset: 'image'
    },
    {
      name: 'buttonLabel',
      title: 'Button Label',
      fieldset: 'buttons',
      type: 'string'
    },
    {
      name: 'buttonLink',
      title: 'Button Link',
      fieldset: 'buttons',
      type: 'url',
      validation: Rule => Rule.uri({ allowRelative: true })
    },
    {
      name: 'drip',
      title: 'Has Drip?',
      fieldset: 'options',
      type: 'boolean'
    },
    {
      name: 'reverse',
      title: 'Reverse Image/Text Arrangement?',
      fieldset: 'options',
      type: 'boolean'
    },
  ]
};
