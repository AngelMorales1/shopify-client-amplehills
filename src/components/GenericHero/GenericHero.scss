@import 'settings', 'typography';

.GenericHero {
  @extend .text-madison-blue;

  &--white {
    @extend .bg-white;
  }

  &--yellow {
    @extend .bg-yellow;
  }

  &--light-yellow {
    @extend .bg-light-yellow;
  }

  &--mustard {
    @extend .bg-deep-yellow;
  }

  &--blue {
    @extend .bg-sky-blue;
  }

  &--light-turquoise {
    @extend .bg-light-turquoise;
  }

  &--pink {
    @extend .bg-light-pink;
  }

  &--light-pink {
    @extend .bg-baby-pink;
  }

  &--light-peach {
    @extend .bg-light-peach;
  }

  &--light-orange {
    @extend .bg-light-orange;
  }

  &--peach-pink {
    @extend .bg-peach-pink;
  }

  &--yellow-green {
    @extend .bg-yellow-green;
  }

  &--light-yellow-green {
    @extend .bg-light-yellow-green;
  }

  &__content-container {
    flex-direction: column;

    @media ($medium-up) {
      flex-direction: row;
      align-items: center;
    }
  }

  &__content-container--reverse {
    @media ($medium-up) {
      flex-direction: row-reverse;
    }
  }

  &__text-container--image-right {
    text-align: center;

    @media ($medium-up) {
      text-align: left;
    }
  }

  &__image {
    max-height: spacing(100);
    object-fit: contain;
  }

  &__button {
    @media ($medium-up) {
      padding-top: calc(#{spacing(4)} + 0.125rem);
      padding-bottom: calc(#{spacing(4)} - 0.125rem);
    }
  }

  &__two-images {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    @media ($medium-up) {
      flex-direction: row;
      justify-content: space-between;
    }
  }
}
