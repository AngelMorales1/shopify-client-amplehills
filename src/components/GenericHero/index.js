import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import cx from 'classnames';

import imageModel from 'models/imageModel';
import styles from './GenericHero.scss';
import { Image, Button } from 'components/base';

const GenericHero = ({ block, z }) => {
  const fields = get(block, 'fields', {});
  const title = get(fields, 'title', '');
  const image1 = get(fields, 'image1', null);
  const isDripOn = get(fields, 'drip', false);
  const colorClass = `GenericHero--${get(fields, 'color', 'blue')}`;
  const text = get(fields, 'text', '');
  const imageRight = get(fields, 'imageRight', false);
  const isReverseArrangement = get(fields, 'isReverseArrangement', false);
  const buttonLabel = get(fields, 'buttonLabel', '');
  const buttonLink = get(fields, 'buttonLink', '');
  const blockHasButton = !!buttonLabel && !!buttonLink;
  const classes = cx(
    styles[colorClass],
    {
      pb4: !isDripOn,
      drip: isDripOn
    },
    'pb2 z-sub-nav'
  );

  return (
    <div style={{ zIndex: z }} className={classes}>
      <div className="transition-slide-up container-width mx-auto pt4 px2 center">
        <p className="block-headline pt3 pb4">{title}</p>
        {image1 ? (
          <Image
            className="col-8 md-col-6 mt4"
            alt={`${title} image`}
            src={contentfulImgUtil(
              get(image1, 'fields.file.url', ''),
              '1400',
              'png'
            )}
          />
        ) : null}
      </div>
    </div>
  );
};

GenericHero.propTypes = {
  z: PropTypes.number,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      color: PropTypes.string,
      drip: PropTypes.bool,
      image1: imageModel.propTypes,
      title: PropTypes.string,
      text: PropTypes.string,
      buttonLink: PropTypes.string,
      buttonLabel: PropTypes.string
    })
  })
};

GenericHero.defaultProps = {
  z: 1,
  block: {
    fields: PropTypes.shape({
      color: 'blue',
      drip: false,
      image1: null,
      title: '',
      text: '',
      buttonLink: '',
      buttonLabel: ''
    })
  }
};

export default GenericHero;
