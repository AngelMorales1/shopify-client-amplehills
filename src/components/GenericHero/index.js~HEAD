import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import cx from 'classnames';

import imageModel from 'models/imageModel';
import styles from './GenericHero.scss';
import { Image } from 'components/base';

const GenericHero = ({ block, z }) => {
  const fields = get(block, 'fields', {});
  const title = get(fields, 'title', '');
  const image1 = get(fields, 'image1', null);
  const isDripOn = get(fields, 'drip', false);
  const colorClass = `GenericHero--${get(fields, 'color', 'blue')}`;
  const classes = cx(
    styles[colorClass],
    {
      drip: isDripOn
    },
    'pb2 z-sub-nav'
  );

  return (
    <div style={{ zIndex: z }} className={classes}>
      <div className="transition-slide-up container-width mx-auto pt4 px2 center">
        <p className="block-headline pt3 pb4">{title}</p>
        {image1 ? (
          <Image
            className="col-8 md-col-6 mt4"
            alt={`${title} image`}
            src={contentfulImgUtil(
              get(image1, 'fields.file.url', ''),
              '1400',
              'png'
            )}
          />
        ) : null}
      </div>
    </div>
  );
};

export default GenericHero;

GenericHero.propTypes = {
  z: PropTypes.number,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      color: PropTypes.string,
      drip: PropTypes.bool,
      image1: imageModel.propTypes,
      title: PropTypes.string
    })
  })
};

GenericHero.defaultProps = {
  z: 1,
  block: {
    fields: PropTypes.shape({
      color: 'blue',
      drip: false,
      image1: null,
      title: ''
    })
  }
};
