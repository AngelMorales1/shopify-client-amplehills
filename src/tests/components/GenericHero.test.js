import React from 'react';
import { shallow } from 'enzyme';
import GenericHero from 'components/GenericHero';
it('renders without data', () => {
  const component = shallow(<GenericHero />);
  expect(component).toMatchSnapshot();
});
it('renders with data', () => {
  const component = shallow(
    <GenericHero
      z={0}
      block={{
        fields: {
          title: 'foo',
          color: 'foo',
          drip: false,
          text: 'foo',
          buttonLabel: 'foo',
          buttonLink: '/foo',
          ImageRight: false,
          isReverseArrangement: false,
          image1: {
            sys: {
              space: {
                sys: {
                  type: 'foo',
                  linkType: 'foo',
                  id: 'foo'
                }
              },
              id: 'foo',
              type: 'foo',
              createdAt: 'foo',
              updatedAt: 'foo',
              environment: { foo: {} },
              revision: 0,
              locale: 'foo'
            },
            image2: {
              sys: {
                space: {
                  sys: {
                    type: 'foo',
                    linkType: 'foo',
                    id: 'foo'
                  }
                },
                id: 'foo',
                type: 'foo',
                createdAt: 'foo',
                updatedAt: 'foo',
                environment: { foo: {} },
                revision: 0,
                locale: 'foo'
              }
            }
          }
        }
      }}
    />
  );
  expect(component).toMatchSnapshot();
});
