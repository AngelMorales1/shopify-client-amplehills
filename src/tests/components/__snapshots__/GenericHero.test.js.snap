// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenericHero
    block={
        Object {
            "fields": Object {
              "ImageRight": false,
              "buttonLabel": "foo",
              "buttonLink": "/foo",
              "color": "foo",
              "drip": false,
              "image1": Object {
                "image2": Object {
                  "sys": Object {
                    "createdAt": "foo",
                    "environment": Object {
                      "foo": Object {},
                    },
                    "id": "foo",
                    "locale": "foo",
                    "revision": 0,
                    "space": Object {
                      "sys": Object {
                        "id": "foo",
                        "linkType": "foo",
                        "type": "foo",
                      },
                    },
                    "type": "foo",
                    "updatedAt": "foo",
                  },
                },
                "sys": Object {
                  "createdAt": "foo",
                  "environment": Object {
                    "foo": Object {},
                  },
                  "id": "foo",
                  "locale": "foo",
                  "revision": 0,
                  "space": Object {
                    "sys": Object {
                      "id": "foo",
                      "linkType": "foo",
                      "type": "foo",
                    },
                  },
                  "type": "foo",
                  "updatedAt": "foo",
                },
              },
              "isReverseArrangement": false,
              "text": "foo",
              "title": "foo",
            },
          }
    }
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex justify-center content-width mx-auto"
>
        <div
                className="col-12 md-col-6 flex justify-center"
        >
                <div
                        className="transition-slide-up py4 px2 col-10 md-col-8 center"
                >
                        <p
                                className="block-headline pt3"
                        >
                                foo
                        </p>
                        <div
                                className="markdown-block pt3"
                                dangerouslySetInnerHTML={
                                        Object {
                                                "__html": "<p>foo</p>
                                        ",
                                              }
                                }
                        />
                        <Image
                                alt="foo image"
                                bg={false}
                                className="w100 pt3"
                                onImgLoad={[Function]}
                                src="?w=1400&fm=png&q=90"
                                style={Object {}}
                        />
                        <div>
                                <Button
                                        childrenWrapperClassName=""
                                        className="inline-flex mt3"
                                        color="madison-blue"
                                        disabled={false}
                                        fullWidth={false}
                                        hover="shadow"
                                        label="foo"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        openInCurrentTab={false}
                                        shadow={false}
                                        to="/foo"
                                        type="button"
                                        variant="primary-responsive"
                                />
                        </div>
                </div>
        </div>
</div>,
      "className": "pb4 pb2 z-sub-nav",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="col-12 md-col-6 flex justify-center"
>
            <div
                        className="transition-slide-up py4 px2 col-10 md-col-8 center"
            >
                        <p
                                    className="block-headline pt3"
                        >
                                    foo
                        </p>
                        <div
                                    className="markdown-block pt3"
                                    dangerouslySetInnerHTML={
                                                Object {
                                                            "__html": "<p>foo</p>
                                                ",
                                                          }
                                    }
                        />
                        <Image
                                    alt="foo image"
                                    bg={false}
                                    className="w100 pt3"
                                    onImgLoad={[Function]}
                                    src="?w=1400&fm=png&q=90"
                                    style={Object {}}
                        />
                        <div>
                                    <Button
                                                childrenWrapperClassName=""
                                                className="inline-flex mt3"
                                                color="madison-blue"
                                                disabled={false}
                                                fullWidth={false}
                                                hover="shadow"
                                                label="foo"
                                                onClick={[Function]}
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                                openInCurrentTab={false}
                                                shadow={false}
                                                to="/foo"
                                                type="button"
                                                variant="primary-responsive"
                                    />
                        </div>
            </div>
</div>,
          null,
          null,
        ],
        "className": "flex justify-center content-width mx-auto",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="transition-slide-up py4 px2 col-10 md-col-8 center"
>
              <p
                            className="block-headline pt3"
              >
                            foo
              </p>
              <div
                            className="markdown-block pt3"
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "<p>foo</p>
                                          ",
                                                      }
                            }
              />
              <Image
                            alt="foo image"
                            bg={false}
                            className="w100 pt3"
                            onImgLoad={[Function]}
                            src="?w=1400&fm=png&q=90"
                            style={Object {}}
              />
              <div>
                            <Button
                                          childrenWrapperClassName=""
                                          className="inline-flex mt3"
                                          color="madison-blue"
                                          disabled={false}
                                          fullWidth={false}
                                          hover="shadow"
                                          label="foo"
                                          onClick={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          openInCurrentTab={false}
                                          shadow={false}
                                          to="/foo"
                                          type="button"
                                          variant="primary-responsive"
                            />
              </div>
</div>,
            "className": "col-12 md-col-6 flex justify-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="block-headline pt3"
>
                  foo
</p>,
                <div
                  className="markdown-block pt3"
                  dangerouslySetInnerHTML={
                                    Object {
                                                      "__html": "<p>foo</p>
                                    ",
                                                    }
                  }
/>,
                <Image
                  alt="foo image"
                  bg={false}
                  className="w100 pt3"
                  onImgLoad={[Function]}
                  src="?w=1400&fm=png&q=90"
                  style={Object {}}
/>,
                <div>
                  <Button
                                    childrenWrapperClassName=""
                                    className="inline-flex mt3"
                                    color="madison-blue"
                                    disabled={false}
                                    fullWidth={false}
                                    hover="shadow"
                                    label="foo"
                                    onClick={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    openInCurrentTab={false}
                                    shadow={false}
                                    to="/foo"
                                    type="button"
                                    variant="primary-responsive"
                  />
</div>,
              ],
              "className": "transition-slide-up py4 px2 col-10 md-col-8 center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "foo",
                  "className": "block-headline pt3",
                },
                "ref": null,
                "rendered": "foo",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "markdown-block pt3",
                  "dangerouslySetInnerHTML": Object {
                    "__html": "<p>foo</p>
",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "foo image",
                  "bg": false,
                  "children": null,
                  "className": "w100 pt3",
                  "onImgLoad": [Function],
                  "src": "?w=1400&fm=png&q=90",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    childrenWrapperClassName=""
                    className="inline-flex mt3"
                    color="madison-blue"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label="foo"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    openInCurrentTab={false}
                    shadow={false}
                    to="/foo"
                    type="button"
                    variant="primary-responsive"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "childrenWrapperClassName": "",
                    "className": "inline-flex mt3",
                    "color": "madison-blue",
                    "disabled": false,
                    "fullWidth": false,
                    "hover": "shadow",
                    "label": "foo",
                    "onClick": [Function],
                    "onMouseEnter": [Function],
                    "onMouseLeave": [Function],
                    "openInCurrentTab": false,
                    "shadow": false,
                    "to": "/foo",
                    "type": "button",
                    "variant": "primary-responsive",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        null,
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex justify-center content-width mx-auto"
>
          <div
                    className="col-12 md-col-6 flex justify-center"
          >
                    <div
                              className="transition-slide-up py4 px2 col-10 md-col-8 center"
                    >
                              <p
                                        className="block-headline pt3"
                              >
                                        foo
                              </p>
                              <div
                                        className="markdown-block pt3"
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": "<p>foo</p>
                                                  ",
                                                          }
                                        }
                              />
                              <Image
                                        alt="foo image"
                                        bg={false}
                                        className="w100 pt3"
                                        onImgLoad={[Function]}
                                        src="?w=1400&fm=png&q=90"
                                        style={Object {}}
                              />
                              <div>
                                        <Button
                                                  childrenWrapperClassName=""
                                                  className="inline-flex mt3"
                                                  color="madison-blue"
                                                  disabled={false}
                                                  fullWidth={false}
                                                  hover="shadow"
                                                  label="foo"
                                                  onClick={[Function]}
                                                  onMouseEnter={[Function]}
                                                  onMouseLeave={[Function]}
                                                  openInCurrentTab={false}
                                                  shadow={false}
                                                  to="/foo"
                                                  type="button"
                                                  variant="primary-responsive"
                                        />
                              </div>
                    </div>
          </div>
</div>,
        "className": "pb4 pb2 z-sub-nav",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <div
              className="col-12 md-col-6 flex justify-center"
>
              <div
                            className="transition-slide-up py4 px2 col-10 md-col-8 center"
              >
                            <p
                                          className="block-headline pt3"
                            >
                                          foo
                            </p>
                            <div
                                          className="markdown-block pt3"
                                          dangerouslySetInnerHTML={
                                                        Object {
                                                                      "__html": "<p>foo</p>
                                                        ",
                                                                    }
                                          }
                            />
                            <Image
                                          alt="foo image"
                                          bg={false}
                                          className="w100 pt3"
                                          onImgLoad={[Function]}
                                          src="?w=1400&fm=png&q=90"
                                          style={Object {}}
                            />
                            <div>
                                          <Button
                                                        childrenWrapperClassName=""
                                                        className="inline-flex mt3"
                                                        color="madison-blue"
                                                        disabled={false}
                                                        fullWidth={false}
                                                        hover="shadow"
                                                        label="foo"
                                                        onClick={[Function]}
                                                        onMouseEnter={[Function]}
                                                        onMouseLeave={[Function]}
                                                        openInCurrentTab={false}
                                                        shadow={false}
                                                        to="/foo"
                                                        type="button"
                                                        variant="primary-responsive"
                                          />
                            </div>
              </div>
</div>,
            null,
            null,
          ],
          "className": "flex justify-center content-width mx-auto",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="transition-slide-up py4 px2 col-10 md-col-8 center"
>
                <p
                                className="block-headline pt3"
                >
                                foo
                </p>
                <div
                                className="markdown-block pt3"
                                dangerouslySetInnerHTML={
                                                Object {
                                                                "__html": "<p>foo</p>
                                                ",
                                                              }
                                }
                />
                <Image
                                alt="foo image"
                                bg={false}
                                className="w100 pt3"
                                onImgLoad={[Function]}
                                src="?w=1400&fm=png&q=90"
                                style={Object {}}
                />
                <div>
                                <Button
                                                childrenWrapperClassName=""
                                                className="inline-flex mt3"
                                                color="madison-blue"
                                                disabled={false}
                                                fullWidth={false}
                                                hover="shadow"
                                                label="foo"
                                                onClick={[Function]}
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                                openInCurrentTab={false}
                                                shadow={false}
                                                to="/foo"
                                                type="button"
                                                variant="primary-responsive"
                                />
                </div>
</div>,
              "className": "col-12 md-col-6 flex justify-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="block-headline pt3"
>
                    foo
</p>,
                  <div
                    className="markdown-block pt3"
                    dangerouslySetInnerHTML={
                                        Object {
                                                            "__html": "<p>foo</p>
                                        ",
                                                          }
                    }
/>,
                  <Image
                    alt="foo image"
                    bg={false}
                    className="w100 pt3"
                    onImgLoad={[Function]}
                    src="?w=1400&fm=png&q=90"
                    style={Object {}}
/>,
                  <div>
                    <Button
                                        childrenWrapperClassName=""
                                        className="inline-flex mt3"
                                        color="madison-blue"
                                        disabled={false}
                                        fullWidth={false}
                                        hover="shadow"
                                        label="foo"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        openInCurrentTab={false}
                                        shadow={false}
                                        to="/foo"
                                        type="button"
                                        variant="primary-responsive"
                    />
</div>,
                ],
                "className": "transition-slide-up py4 px2 col-10 md-col-8 center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "foo",
                    "className": "block-headline pt3",
                  },
                  "ref": null,
                  "rendered": "foo",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "markdown-block pt3",
                    "dangerouslySetInnerHTML": Object {
                      "__html": "<p>foo</p>
",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "alt": "foo image",
                    "bg": false,
                    "children": null,
                    "className": "w100 pt3",
                    "onImgLoad": [Function],
                    "src": "?w=1400&fm=png&q=90",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Button
                      childrenWrapperClassName=""
                      className="inline-flex mt3"
                      color="madison-blue"
                      disabled={false}
                      fullWidth={false}
                      hover="shadow"
                      label="foo"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      openInCurrentTab={false}
                      shadow={false}
                      to="/foo"
                      type="button"
                      variant="primary-responsive"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": null,
                      "childrenWrapperClassName": "",
                      "className": "inline-flex mt3",
                      "color": "madison-blue",
                      "disabled": false,
                      "fullWidth": false,
                      "hover": "shadow",
                      "label": "foo",
                      "onClick": [Function],
                      "onMouseEnter": [Function],
                      "onMouseLeave": [Function],
                      "openInCurrentTab": false,
                      "shadow": false,
                      "to": "/foo",
                      "type": "button",
                      "variant": "primary-responsive",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          null,
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenericHero
    block={
        Object {
            "fields": Object {
              "buttonLabel": "",
              "buttonLink": "",
              "color": "blue",
              "drip": false,
              "image1": null,
              "text": "",
              "title": "",
            },
          }
    }
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex justify-center content-width mx-auto"
>
        <div
                className="col-12 md-col-6 flex justify-center"
        >
                <div
                        className="transition-slide-up py4 px2 col-10 md-col-8 center"
                >
                        <p
                                className="block-headline pt3"
                        />
                </div>
        </div>
</div>,
      "className": "pb4 pb2 z-sub-nav",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="col-12 md-col-6 flex justify-center"
>
            <div
                        className="transition-slide-up py4 px2 col-10 md-col-8 center"
            >
                        <p
                                    className="block-headline pt3"
                        />
            </div>
</div>,
          null,
          null,
        ],
        "className": "flex justify-center content-width mx-auto",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="transition-slide-up py4 px2 col-10 md-col-8 center"
>
              <p
                            className="block-headline pt3"
              />
</div>,
            "className": "col-12 md-col-6 flex justify-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="block-headline pt3"
/>,
                null,
                null,
                null,
              ],
              "className": "transition-slide-up py4 px2 col-10 md-col-8 center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "block-headline pt3",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              null,
              null,
              null,
            ],
            "type": "div",
          },
          "type": "div",
        },
        null,
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex justify-center content-width mx-auto"
>
          <div
                    className="col-12 md-col-6 flex justify-center"
          >
                    <div
                              className="transition-slide-up py4 px2 col-10 md-col-8 center"
                    >
                              <p
                                        className="block-headline pt3"
                              />
                    </div>
          </div>
</div>,
        "className": "pb4 pb2 z-sub-nav",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <div
              className="col-12 md-col-6 flex justify-center"
>
              <div
                            className="transition-slide-up py4 px2 col-10 md-col-8 center"
              >
                            <p
                                          className="block-headline pt3"
                            />
              </div>
</div>,
            null,
            null,
          ],
          "className": "flex justify-center content-width mx-auto",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="transition-slide-up py4 px2 col-10 md-col-8 center"
>
                <p
                                className="block-headline pt3"
                />
</div>,
              "className": "col-12 md-col-6 flex justify-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="block-headline pt3"
/>,
                  null,
                  null,
                  null,
                ],
                "className": "transition-slide-up py4 px2 col-10 md-col-8 center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "block-headline pt3",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
                null,
                null,
                null,
              ],
              "type": "div",
            },
            "type": "div",
          },
          null,
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
